packages/main/src/modules/bin.ts

```diff
@@ -170,6 +171,49 @@ export async function install() {
   }
 }
 
+export async function npmPackageOutdated(projectPath: string, packageName: string = '') {
+  const normalizedPath = path.normalize(projectPath);
+  const nodeModulesPath = path.join(normalizedPath, 'node_modules');
+
+  if (!existsSync(nodeModulesPath)) {
+    // For projects without node_modules, install dependencies before checking for outdated packages
+    await installNpmPackages(projectPath);
+  }
+
+  log.info(`[PackageOutdated] Verifying if the package ${packageName} is outdated...`);
+
+  try {
+    const _npmPackageOutdated = run('npm', 'npm', {
+      args: ['outdated', packageName, '--depth=0', '--json'],
+      cwd: projectPath,
+    });
+    log.info(`[PackageOutdated] Package ${packageName} is outdated...`);
+    await _npmPackageOutdated.waitFor(new RegExp(packageName), /\{\}/);
+    return true;
+  } catch (_) {
+    log.info(`[PackageOutdated] Package ${packageName} is up to date...${normalizedPath}`);
+    return false;
+  }
+}
+
+export async function installNpmPackages(projectPath: string, packageName?: string) {
```

Comment:
I think we should move this and the above to their own module, like `modules/npm.ts`, and have them there as `outdated` and `install`. 

The `bin` module is more generic for running node binaries and these function are very npm-specific.